meja/tests/records.ml

```diff
@@ -40,30 +56,46 @@ module X = struct
   include struct
     type 'a t = {a: 'a; b: 'a; c: 'a}
 
-    let typ : (_, _ t) Typ.t =
+    let typ __implicit24__ __implicit23__ __implicit22__ __implicit21__
```

Comment:
Shouldn't this just have one implicit argument?

Reply:
Ah I guess this is fixed in https://github.com/o1-labs/snarky/pull/256 ?

Reply:
Not quite, #256 is part of the fix, but I'm not quite finished with the unification changes to make it work properly.

I've spun this out as its own PR because the old version was fragile and needed updating with nearly every typechecker change, whereas the new version generates a parsetree and lets the typechecker handle it as normal.